src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnection.cs

```diff
@@ -297,8 +315,54 @@ public static void RegisterColumnEncryptionKeyStoreProviders(IDictionary<string,
                     new Dictionary<string, SqlColumnEncryptionKeyStoreProvider>(customProviders, StringComparer.OrdinalIgnoreCase);
 
                 // Set the dictionary to the ReadOnly dictionary.
-                _CustomColumnEncryptionKeyStoreProviders = new ReadOnlyDictionary<string, SqlColumnEncryptionKeyStoreProvider>(customColumnEncryptionKeyStoreProviders);
+                _GlobalCustomColumnEncryptionKeyStoreProviders = new ReadOnlyDictionary<string, SqlColumnEncryptionKeyStoreProvider>(customColumnEncryptionKeyStoreProviders);
+            }
+        }
+
+        /// <include file='../../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlConnection.xml' path='docs/members[@name="SqlConnection"]/RegisterColumnEncryptionKeyStoreProvidersOnConnection/*' />
+        public void RegisterColumnEncryptionKeyStoreProvidersOnConnection(IDictionary<string, SqlColumnEncryptionKeyStoreProvider> customProviders)
```

Comment:
Not sure what the best name would be. Suggestions welcome.

Reply:
cc @David-Engel

Reply:
I think RegisterConnectionColumnEncryptionKeyStoreProviders would be better. It is kind of long and awkward either way.

Since it's a new public API, I'd get @saurabh500 's opinion, too. We are adding a SqlConnection instance-specific key store provider dictionary that can override the static/global one that is set via SqlConnection.RegisterColumnEncryptionKeyStoreProviders().

Reply:
Also tagging @Xtrimmer @jaszymas @raghav131 for feedback on the new API name.

Reply:
I'm going going to change my vote to Johnny's original proposal. Given that we can't change the existing method (which would be ideal, IMO) and that we likely need to add registration at the SqlCommand level, too, RegisterColumnEncryptionKeyStoreProvidersOnConnection and RegisterColumnEncryptionKeyStoreProvidersOnCommand probably make more sense together than my other proposal.